### PR TITLE
Add (_)dummy_threading module, identical to threading

### PR DESCRIPTION
**Pull request by CraftSpider**
As in title. Note that I made `dummy_threading` just re-export `_dummy_threading`, as it does in real code. Can't be *exactly* identical to threading, as it includes a `__all__` not in threading, unless `__all__` should be added to threading (Wasn't sure if that was in scope)

### CODE REVIEW

File: stdlib/2and3/dummy_threading.pyi [hunk 1]
```diff
@@ -0,0 +1,43 @@
+
+import sys
+
+from _dummy_threading import (
```
Comment (JelleZijlstra) [inline stdlib/2and3/dummy_threading.pyi]: Just `from _dummy_threading import *` is fine.

File: stdlib/2and3/_dummy_threading.pyi [hunk 2]
```diff
@@ -0,0 +1,198 @@
+# Stubs for threading
```
Comment (JelleZijlstra) [inline stdlib/2and3/_dummy_threading.pyi]: No need for this comment

Comment (CraftSpider) [reply stdlib/2and3/_dummy_threading.pyi]: Should I remove it from `threading.pyi`? As they're going to be checked consistent

**Comment by CraftSpider** [comment]
Removed the comment, ready for re-review

File: stdlib/2and3/dummy_threading.pyi [hunk 3]
```diff
@@ -0,0 +1,3 @@
+
```
Comment (asottile) [inline stdlib/2and3/dummy_threading.pyi]: this module (and _dummy_threading) is removed in python3.9 -- does typeshed have a way to represent that?

Comment (JelleZijlstra) [reply stdlib/2and3/dummy_threading.pyi]: No, see #16

**Comment by CraftSpider** [comment]
@JelleZijlstra is there anything blocking this being merged at this point?

**Pull request merged**